extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -43,7 +43,17 @@ class AmpBaseCarousel extends PreactBaseElement {
       },
       ActionTrust.LOW
     );
+    const mu = new MutationObserver(() => {
```

Comment:
No need for a separate observer. You can just override `checkPropsPostMutations` method.

Reply:
Is it preferred to override `checkPropsPostMutations` to check for one prop mutation? It seems imprecise to use code that runs any time any of the prop attributes change, especially since `slide` (which is not needed in the component props def) would have to be added to this list only to trigger this callback. It seems like a separate observer is more appropriate for this case?

Reply:
[This would be the associated change](https://github.com/ampproject/amphtml/pull/31318/commits/6a249f78a72e9110ea8399a10aa060c86035b697)

Reply:
Yeah. This is a bit messy. But it'd be nice to only use one mutation observer. An alternative I see is to modify [this method](https://github.com/ampproject/amphtml/blob/35948a99954ff647ad38eff50a802e473d3c45f1/src/preact/base-element.js#L399) slightly with an additional callback, e.g. `mutationObserverCallback()` or such. Or we could even make `checkMutations` itself protected and overridable?

Reply:
I don't think we should make `checkMutations` overridable given that components should still rerender if their attributes in the prop def mutate. I modified it to also call a `mutationObserverCallback()` with the records, however.